llvm/lib/Target/DirectX/DXILPostOptimizationValidation.cpp

```diff
@@ -286,11 +308,13 @@ class DXILPostOptimizationValidationLegacy : public ModulePass {
     AU.addRequired<DXILResourceBindingWrapperPass>();
     AU.addRequired<DXILMetadataAnalysisWrapperPass>();
     AU.addRequired<RootSignatureAnalysisWrapper>();
+    AU.addRequired<DXILResourceTypeWrapperPass>();
     AU.addPreserved<DXILResourceWrapperPass>();
     AU.addPreserved<DXILResourceBindingWrapperPass>();
     AU.addPreserved<DXILMetadataAnalysisWrapperPass>();
     AU.addPreserved<ShaderFlagsAnalysisWrapper>();
     AU.addPreserved<RootSignatureAnalysisWrapper>();
+    AU.addPreserved<DXILResourceTypeWrapperPass>();
```

Comment:
`DXILResourceTypeWrapperPass` is an `ImmutablePass` - it doesn't need to be preserved.